_posts/2019-01-17-multi-store-custom-telemetry-with-shared-data.md

```diff
@@ -0,0 +1,188 @@
+The [Telemetry component](https://searchfox.org/mozilla-central/source/toolkit/components/telemetry/) provides APIs used by the various other parts to gather this data
+and is responsible for storing, collecting and sending this data in what we call "pings", a periodic collection of measurements.
+Telemetry data is only ever send out if the user agreed to it (see "Data Collection and Use" in the "Privacy & Security" preferences of your Firefox).
+
+Data can be collected in one of 3 different formats: histograms, scalars & events (see [our collection overview](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/index.html)).
+Firefox sends this data in the ["main" ping](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/main-ping.html) once in a while (usually roughly daily) and clears out the stored data locally.
+A "main" ping always corresponds to a subsession, which itself is part of a session, this is further explained [in our Session concepts](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/concepts/sessions.html).
+
+People working with the data can therefore make certain assumptions on how to interpret the data from multiple pings across sessions and subsessions.
+
+### The problem
+
+Some data should not be correlated with other data due to privacy concerns.
+So far we had to push Telemetry users to create custom pings and keep track of their own data.
+If they rely on scalars or histograms as recorded my Telemetry, but send a custom ping in different intervals, they can't make valid assumptions about the metrics, as they might have been reset in between.
+This leads to weird hacks or unnecessary code duplication. Additionally we can't provide any help or support for custom data and our tools can't handle it automatically (e.g. to generate dashboards).
+
+### A solution
+
+**Multi-Store**.
+
+Every metric was always tied to the schedule of the "main" ping (read: subsession/session).
+Our **multi-store** solution now enables metrics to be associated with multiple stores at once, defaulting to be in the "main" ping only.
+Any user with custom requirements can now select metrics to be included in their custom store (which is still subject to [Data Collection Review](https://wiki.mozilla.org/Firefox/Data_Collection)).
+Telemetry is still responsible for actual data storage and the APIs, but now the custom ping is responsible for collecting, clearing and periodically sending this data.
+
+### An example
+
+To demonstrate how this is used, let's create a custom ping, which will include one metric of its own and one that's also available in the "main" store (and thus the "main" ping).
+
+We start by adding a new metric to [Histograms.json](https://searchfox.org/mozilla-central/source/toolkit/components/telemetry/Histograms.json):
```

Comment:
One totally optional thought:
Could we use a scalar as an example, to shift the attention a bit away from histograms?